src/editable.ts

```diff
@@ -0,0 +1,328 @@
+        const dataId = getDataId(gateElem);
+        gateElem.append(createLeftDropzone(x, y, height, dataId));
+        gateElem.append(createRightDropzone(x, y, width, height, dataId));
+    });
+};
+
+const addDocumentEvents = (container: HTMLElement): void => {
+    container.addEventListener('click', (ev: MouseEvent) => {
+        _sourceTarget = null;
+        if (ev.ctrlKey) return;
+    });
+    container.addEventListener('contextmenu', (ev: MouseEvent) => {
+        ev.preventDefault();
+    });
+    container.addEventListener('mouseup', () => {
+        cursorCopy(container, false);
+        cursorMove(container, false);
+    });
+};
+
+const addDropzoneEvents = (context: Context): void => {
+    const { container } = context;
+    const dropzoneElems = container.querySelectorAll<SVGRectElement>('.dropzone');
+    dropzoneElems.forEach((dropzoneElem) => {
+        dropzoneElem.addEventListener('mouseup', (ev: MouseEvent) => handleDropzoneMouseUp(ev, context));
+    });
+};
+
+const addMouseEvents = (context: Context): void => {
+    const { container } = context;
+    const gateElems = getGateElems(container);
+    gateElems.forEach((gateElem) => {
+        gateElem.addEventListener('mousedown', (ev: MouseEvent) => handleGateMouseDown(ev, container));
+    });
+};
+
+// Event handlers
+const handleGateMouseDown = (ev: MouseEvent, container: HTMLElement): void => {
+    ev.stopPropagation();
+    _sourceTarget = ev.currentTarget as SVGGElement;
+
+    // Ctrl + Mousedown to copy. Mousedown only to move.
+    ev.ctrlKey ? cursorCopy(container, true) : cursorMove(container, true);
+};
+
+const handleDropzoneMouseUp = (ev: MouseEvent, context: Context): void | false => {
+    ev.stopPropagation();
+
+    const { container, operations, wires, renderFn } = context;
+
+    const currentTarget = ev.currentTarget as SVGGElement;
+
+    if (!currentTarget) return false;
+
+    const dataId = getDataId(currentTarget);
+    const parent = getParent(dataId, operations);
+    const index = splitDataId(dataId).pop();
+    const position = getDropzonePosition(currentTarget);
+
+    if (_sourceTarget == null) return false;
+
+    const sourceDataId = getDataId(_sourceTarget);
+    const sourceParent = getParent(sourceDataId, operations);
+    const sourceIndex = splitDataId(sourceDataId).pop();
+
+    if (index == null || sourceIndex == null) return false;
+
+    const newGate = getGate(sourceDataId, operations);
+    const wireY = getClosestWireY(ev.offsetY, wires);
+
+    // Not allow Measure gate to move vertically
+    if (wireY != null && newGate.gate !== 'measure') {
+        // wires[wireY] returns qubit name (i.e: 'q0')
+        // this remove 'q' and assign an index (i.e: 0)
+        const index = Number(wires[wireY].slice(1));
+        const [firstTarget, ...targetsExceptFirst] = newGate.targets;
+        // Reserve all other properties, only change qId
+        Object.assign(firstTarget, { ...firstTarget, qId: index });
```

Comment:
This update should be done for all targets. Something like for each nth target, find the offset between the original firstTarget.qId and nTarget.qId, then update nTarget.qId to index + offset. Then newGate should be updated with all of these new targets in L129. The gates controls should be updated similarly.